model/pod/db_test.go

```diff
@@ -1 +1,41 @@
 package pod
+
+import (
+	"testing"
+
+	"github.com/evergreen-ci/evergreen/db"
+	"github.com/evergreen-ci/utility"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestFindByInitializing(t *testing.T) {
+	require.NoError(t, db.Clear(Collection))
+
+	p := &Pod{
+		ID:     utility.RandomString(),
+		Status: StatusInitializing,
+		Secret: "secret",
+	}
+	require.NoError(t, p.Insert())
+
+	p = &Pod{
+		ID:     utility.RandomString(),
+		Status: StatusStarting,
+		Secret: "secret",
+	}
+	require.NoError(t, p.Insert())
+
+	p = &Pod{
+		ID:     utility.RandomString(),
+		Status: StatusInitializing,
+		Secret: "secret",
+	}
+	require.NoError(t, p.Insert())
+
+	pods, err := FindByInitializing()
+	require.NoError(t, err)
+	require.Len(t, pods, 2)
+	assert.Equal(t, StatusInitializing, pods[0].Status)
+	assert.Equal(t, StatusInitializing, pods[1].Status)
```

Comment:
This should also check that the two pods you were returned have the same IDs as the ones you expected to rule out some weird possibility that somehow the starting pod ID was returned.

Reply:
I wouldn't do it this way because the DB doesn't promise that it'll be returned in any particular order unless you sort the results. I would check them in a set-like way (e.g. iterate through the DB results, verify that each pod ID in the set of expected ones is found).